lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs

```diff
@@ -751,14 +757,42 @@ spec = describe "SHELLEY_WALLETS" $ do
             ru <- request @ApiWallet ctx ("PUT", endpoint) headers newName
             verify ru expectations
 
-    it "WALLETS_UPDATE_PASS_01 - passphaseLastUpdate gets updated" $ \ctx -> runResourceT $ do
+    it "WALLETS_UPDATE_PASS_01a - passphraseLastUpdate gets updated"
+      $ \ctx -> runResourceT $ do
         w <- emptyWallet ctx
         let payload = updatePassPayload fixturePassphrase "New passphrase"
         let endpoint = "v2/wallets" </> (w ^. walletId)
                 </> ("passphrase" :: Text)
         rup <- request @ApiWallet ctx ("PUT", endpoint) Default payload
         expectResponseCode HTTP.status204 rup
+        let getEndpoint = "v2/wallets" </> (w ^. walletId)
+        let originalPassUpdateDateTime = w ^. #passphrase
+        rg <- request @ApiWallet ctx ("GET", getEndpoint) Default Empty
+        expectField #passphrase (`shouldNotBe` originalPassUpdateDateTime) rg
 
+    it "WALLETS_UPDATE_PASS_01b - passphraseLastUpdate gets updated, mnemonic"
+      $ \ctx -> runResourceT $ do
+        (w, mnemonic) <- emptyWalletAndMnemonic ctx
+        let payload = updatePassPayloadMnemonic mnemonic "New passphrase"
+        let endpoint = "v2/wallets" </> (w ^. walletId)
+                </> ("passphrase" :: Text)
+        rup <- request @ApiWallet ctx ("PUT", endpoint) Default payload
+        expectResponseCode HTTP.status204 rup
+        let getEndpoint = "v2/wallets" </> (w ^. walletId)
+        let originalPassUpdateDateTime = w ^. #passphrase
```

Comment:
(
```suggestion
            originalPassUpdateDateTime = w ^. #passphrase
```

is also possible because it's in the scope of the previous `let`. Both options are perfectly fine, just wanted to point out the possiblity.)

Reply:
As I mention to Yura, I'm inclined not to change code that is there already. But I agree with you multiple lets can be squashed for better readability.